docs/latex/Inhalt/03_Inhalt/02_ddd.tex

```diff
@@ -0,0 +1,35 @@
+
+\paragraph{Wesentliche Begriffe} Da die Domäne der Applikation im Wesentlichen die Themen Kochrezepte und Lebensmittel betrifft und es sich hierbei um Gebiet handelt, welches auch in der englischen Sprache präsent ist, können für die zentralen Begriffe der Domäne die im Englischen üblichen Begriffe verwendet werden. 
+
+Ein zentrales Konzept der Domäne ist das Kochrezept. Ein Kochrezept wird in der Ubiquitous Language mit dem Begriff \enquote{Recipe} bezeichnet. Ein Recipe besitzt typischerweise eine Liste von Zutaten, welche in der Domäne als \enquote{Ingredient} bezeichnet werden. Eine Ingredient bezeichnet ein Nahrungsmittel, von dem für ein Rezept eine bestimmte Menge verwendet wird, welche in einer bestimmten Einheit gemessen wird. Dieses Nahrungsmittel wird als \enquote{Food} bezeichnet, die verwendete Menge als \enquote{Quantity} und die entsprechende Einheit als \enquote{Unit}.
+
+Das zweite zentrale Konzept der Domäne ist das Benutzerprofil, welches in der Ubiquitous Language als \enquote{Profile} bezeichnet wird. Zu einem Profile gehört, wie in der Einleitung erwähnt, eine Menge vorrätiger Foods. Neben den vorrätigen Foods gehören zu einem Profile auch die Meinungen des Nutzers zu bestimmten Lebensmitteln, welche als \enquote{Opinions} bezeichnet werden. Diese Opinion des Nutzers kann sieben Stufen annehmen: \enquote{Foodgasm}, \enquote{Love}, \enquote{Like}, \enquote{Indifferent}, \enquote{Dislike}, \enquote{Hate} und \enquote{Dealbeaker}, hier in zunehmender Reihenfolge der Abneigung des Nutzers gegenüber dem betreffenden Lebensmittel.
+
+\section{Entities \& Value Objects}
+
+\begin{figure}[ht!]
+    \includegraphics[width=0.98\columnwidth]{../diagrams/entity_uml.pdf}
+    \caption{Klassendiagramm der Domäne}
+    \label{fig:class-diag-domain}
+\end{figure}
+
+\autoref{fig:class-diag-domain} zeigt das Klassendiagramm der Domäne. Im Package recipe existiert eine Entity: das Recipe. Jedes Recipe besitzt durch eine UUID eine eigene Identität. Außerdem können Rezepte in seltenen Fällen auch nachträglich abgewandelt werden und besitzen daher grundsätzlich veränderliche Werte. Bei der Ingredient hingegen handelt es sich um ein Value Object, da diese keine eigene Identität besitzt. Existieren zwei Ingredients mit dem gleichen Food und der gleichen Quantity, so sind diese gleich. Außerdem ist ihr Zustand unveränderlich, so wird etwa bei einer Änderung in einem Rezept die betreffende Ingredient gelöscht und eine neue erstellt. Ebenso verhält es sich mit der Quantity: Sind Amount und Unit einer Quantity gleich, so sind auch die Quantities gleich und auch ihr Zustand ist unveränderlich. Zuletzt enthält das Package noch die Unit. Auch hier handelt es sich um ein Value Object, da auch die Gleichheit von Units lediglich von ihren Werten, also von ihren Namen abhängt. Ebenso besitzt eine Unit keinen eigenen Lebenszyklus und ihr Zustand ist unveränderlich.
```

Comment:
"Package" => Paket?
nvm, wir reden ja auch von Entity und Value Object

Reply:
"eine Entity: das Recipe" => "eine Entity: Das Recipe"

Reply:
Klassennamen, packagenamen bitte in \code

Reply:
Wenn nach einem Doppelpunkt kein vollständiger Satz kommt schreibt man klein weiter. Das andere ist korrigiert.